src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -332,6 +333,61 @@ def devices(self, svc_id):
         # (str) -> dict
         return CephService.send_command('mon', 'device ls-by-daemon', who='osd.{}'.format(svc_id))
 
+    def _merge_with_inventory(self, host_previews):
+        """Merge previews with inventory."""
+        osd_previews = []
+        orch = OrchClient.instance()
+        hosts = [host_preview['host'] for host_preview in host_previews]
+        inventory_nodes = [node.to_json() for node in orch.inventory.list(
```

Comment:
@bk201 So it seems the bottleneck comes from here... Are there any chances to improve the latency/scaling of this method, @sebastian-philipp ?